src/primitives/cone.js

```diff
@@ -0,0 +1,265 @@
+	const [conesBegin, conesEnd] = get2PopulatedCoordinateBuffers(coords, extent);
+
+	const coneGeometry = new InstancedBufferGeometry().copy(
+		new CylinderGeometry(radius, 0, 1, 24)
+			.translate(0, -0.5, 0) // translate the geometry so we don't need to calculate the middle of each coordinates-pair
+			.rotateX(Math.PI / 2) // rotate the cone 90 degrees to lookAt work
+	);
+
+	coneGeometry.instanceCount = coords.length / 2;
+
+	coneGeometry.setAttribute(
+		'coneBegin',
+		new InstancedBufferAttribute(conesBegin, 3)
+	);
+
+	coneGeometry.setAttribute(
+		'coneEnd',
+		new InstancedBufferAttribute(conesEnd, 3)
+	);
+
+	const cones = new Mesh(
+		coneGeometry,
+		new ShaderMaterial({
+			transparent: opacity !== 1,
+			depthWrite: opacity === 1,
+			lights: true,
+			uniforms: {
+				...UniformsLib.lights,
+				diffuse: { value: color },
+				opacity: { value: opacity }
+			},
+			vertexShader: `
+				attribute vec3 coneBegin;
+				attribute vec3 coneEnd;
+
+				varying vec3 vLightFront;
+				varying vec3 vIndirectFront;
+
+				#include <common>
+				#include <bsdfs>
+				#include <lights_pars_begin>
+
+				void main() {
+					vec3 z = normalize(coneBegin - coneEnd);
+					// if z.z is 0 the cone doesn't appear
+					z.z += 0.0001;
+
+					vec3 x = normalize(cross(vec3(0, 1, 0), z));
+					vec3 y = cross(z, x);
+
+					float height = distance(coneBegin, coneEnd);
+
+					// position, rotate and scale the cone
+					mat4 coneMatrix = mat4(
+						x, 0,            // row 0
+						y, 0,            // row 1
+						z * height, 0,   // row 2
+						coneBegin, 1 // row 3
+					);
+
+					vec4 mvPosition = modelViewMatrix * coneMatrix * vec4(position, 1);
+
+					gl_Position = projectionMatrix * mvPosition;
+
+					vec3 transformedNormal = normalMatrix * normal;
+
+					#include <lights_lambert_vertex>
+				}
+			`,
+			fragmentShader: `
+				uniform vec3 diffuse;
+				uniform float opacity;
+
+				varying vec3 vLightFront;
+				varying vec3 vIndirectFront;
+
+				#include <common>
+				#include <bsdfs>
+
+				void main() {
+					gl_FragColor = vec4(
+						(vLightFront + vIndirectFront) * BRDF_Lambert(diffuse),
+						opacity
+					);
+				}
+			`
+		})
+	);
+
+	cones.frustumCulled = false;
+
+	if (edgeForm.showEdges === false) {
+		// If the edges aren't shown the work is done.
+		return cones;
+	}
+
+	const group = new Group();
+
+	group.add(cones);
+
+	// Differently from cuboid's edges, the cones' ones are in a different object. It is very hard or maybe impossible to draw edges with complex shapes in the fragment shader.
+
+	// The lines below are the edges' vertices' positions.
+	// The magic numbers below are modified from the position attribute of a three.js EdgesGeometry of the cone.
+	// To get them: console.log(new EdgesGeometry(coneGeometry).attributes.position.array)
+
+	const edgesGeometry = new InstancedBufferGeometry()
+		.setAttribute(
+			'position',
+			new BufferAttribute(
+				new Float32Array([
+					0, -1, 0,
```

Comment:
What's a breakout of the structure of these coordinates?

Reply:
This is the cone base, I changed the variables' names to make this more clear, and changed a comment.